tools/test/h5repack/h5repacktst.c

```
@@ -1592,17 +1594,36 @@ main(void)
     /* Remove test files */
     TESTING("    test file cleanup");
 
-    for (size_t i = 0; i < NELMTS(H5REPACK_TEST_H5_FILES); i++) {
-        if (H5Fdelete(H5REPACK_TEST_H5_FILES[i], H5P_DEFAULT) < 0) {
-            printf(" Failed to delete %s", H5REPACK_TEST_H5_FILES[i]);
-            GOERROR;
-        }
+    if ((fapl_id = h5_fileaccess()) < 0) {
+        printf(" Failed to generate FAPL");
+        GOERROR;
     }
+
+    h5_delete_all_test_files(H5REPACK_TEST_H5_FILES, fapl_id);
+
     /* Also clean up H5REPACK_FSPACE_FNAMES array files */
     for (size_t i = 0; i < NELMTS(H5REPACK_FSPACE_FNAMES); i++) {
-        if (H5Fdelete(H5REPACK_FSPACE_FNAMES[i], H5P_DEFAULT) < 0) {
-            printf(" Failed to delete %s", H5REPACK_FSPACE_FNAMES[i]);
-            GOERROR;
+        /* Only the files at these indices are unconditionally created */
+        if (i == 0 || i == 1 || i == 2) {
```

This is going to be fragile at best since new filenames aren't always inserted at the end of lists and often end up somewhere in the middle. It seems that rather than trying to cleanup files at the end and guessing which ones are there, it would be better to either ignore failures (and enclose these in `H5E_BEGIN_TRY`/`H5E_END_TRY` blocks), or separate these into sub-functions that are responsible for cleaning up their own test files.

---

`h5_delete_test_file()` already ignores cleanup failures, so this check was unnecessary in the first place - I've merged the two blocks that clean up `H5REPACK_FSPACE_FNAMES`.